README.md

```diff
@@ -111,6 +131,16 @@ The value contains the following elements:
 
 Provides a friendly name for the service used for identification in the user interface.
 
+###### AuthenticationMethod
+
+Specifies the type of authentication the service will use, from one of the following available options: `OAuth1`, `OAuth2AuthorizationCode`, `OAuth2ClientCredentials` or `ApiKey`.
+
+If none provided, it will default to `OAuth2AuthorizationCode`.
+
+###### ClientCredentialsProvision
+
+Specifies the available options for providing credentials in an `OAuth2` flow: `AuthHeader` or `RequestBody`.
```

Comment:
Think we should add something here about what `AuthenticatedMethod` this is relevant for.
```suggestion
Specifies the available options for providing credentials in an `OAuth2` flow: `AuthHeader` or `RequestBody`.

This setting is only utilized when the `AuthenticatedMethod` value is configured as `OAuth2ClientCredentials`.
```